google/cloud/forseti/services/dao.py

```diff
@@ -258,51 +258,174 @@ def define_model(model_name, dbengine, model_seed):
+                msg = binascii.hexlify(serialized_ctr)
+                msg += self.full_name
+                return hmac.new(model_seed.encode('utf-8'), msg).hexdigest()
 
-            Returns:
-                str: etag to avoid race condition when set policy
-            """
-            serialized_ctr = struct.pack('>I', self.policy_update_counter)
-            msg = binascii.hexlify(serialized_ctr)
-            msg += self.full_name
-            return hmac.new(model_seed.encode('utf-8'), msg).hexdigest()
+            def __repr__(self):
+                """String representation.
 
-        def __repr__(self):
-            """String representation.
+                Returns:
+                    str: Resource represented as
+                        (full_name='{}', name='{}' type='{}')
+                """
+                return '<Resource(full_name={}, name={} type={})>'.format(
+                    self.full_name, self.name, self.type)
 
-            Returns:
-                str: Resource represented as
-                    (full_name='{}', name='{}' type='{}')
-            """
-            return '<Resource(full_name={}, name={} type={})>'.format(
-                self.full_name, self.name, self.type)
+        class Binding(base):
+            """Row for a binding between resource, roles and members."""
+
+            __tablename__ = bindings_tablename
+            id = Column(Integer, Sequence('{}_id_seq'.format(bindings_tablename)),
+                        primary_key=True)
+            resource_type_name = Column(
+                String(512),
+                ForeignKey('{}.type_name'.format(resources_tablename)))
+            role_name = Column(String(128), ForeignKey(
+                '{}.name'.format(roles_tablename)))
+
+            resource = relationship('Resource', remote_side=[resource_type_name])
+            role = relationship('Role', remote_side=[role_name])
+
+            members = relationship('Member',
+                                   secondary=binding_members,
+                                   back_populates='bindings')
+
+            def __repr__(self):
+                """String Representation
+
+                Returns:
+                    str: Binding represented as
+                        (id='{}', role='{}', resource='{}' members='{}')
+                """
+                fmt_s = '<Binding(id={}, role={}, resource={} members={})>'
+                return fmt_s.format(
+                    self.id,
+                    self.role_name,
+                    self.resource_type_name,
+                    self.members)
+
+    else:
+        class Resource(base):
+            """Row entry for a GCP resource."""
+            __tablename__ = resources_tablename
+
+            full_name = Column(String(2048), nullable=False)
+            # The collation type is used here to allow the String column
+            # to be case sensitive.
+            type_name = Column(String(512, collation='utf8mb4_bin'),
```

Comment:
Would something like this work?
```
type_name = Column(get_string(), primary_key=True)
```
And it is inside `get_string()` that you have the logic to determine whether a `String()` is returned with the collation attribute or not?

Might also want to be able to passing in a parameter to determine the size, eg. 512